store/models/run.go

```diff
@@ -189,13 +189,23 @@ func (rr RunResult) GetError() error {
 	}
 }
 
-// MergeData merges the existing Data on a RunResult with the given JSON.
-func (rr RunResult) MergeData(j JSON) (RunResult, error) {
-	merged, err := rr.Data.Merge(j)
+// Merge merges the input RunResult with the instance it is called on,
+// preferring the RunResult values passed in, but using the existing values
+// if the input RunResult values are of their respective zero value.
+func (rr RunResult) Merge(in RunResult) (RunResult, error) {
+	merged, err := rr.Data.Merge(in.Data)
 	if err != nil {
-		return rr, fmt.Errorf("TaskRun#Merge merging JSON: %v", err.Error())
+		return in, fmt.Errorf("TaskRun#Merge merging JSON: %v", err.Error())
 	}
-
-	rr.Data = merged
-	return rr, nil
+	in.Data = merged
+	if !in.ErrorMessage.Valid && rr.ErrorMessage.Valid {
+		in.ErrorMessage = rr.ErrorMessage
+	}
+	if len(in.JobRunID) == 0 {
+		in.JobRunID = rr.JobRunID
+	}
+	if in.Pending || rr.Pending {
+		in.Pending = true
```

Comment:
If any of them are `Pending`, the result is `pending`. How do you then transition a `RunResult` off pending? Explicitly through the JobRunner?

I would stick to the description and just delete this whole clause since `in` will have the pending we want.

Reply:
The adapter resumes a pending run and figures out when to transition off of pending. The run result needs to be marked pending when it's passed in so that the adapter knows it is resuming and not starting fresh.